tests/e2e/tidbcluster/serial.go

```diff
@@ -589,6 +579,213 @@ var _ = ginkgo.Describe("[Serial]", func() {
 		})
 	})
 
+	ginkgo.Describe("Canary Deploy TiDB Operator", func() {
+		var oa *tests.OperatorActions
+		var ocfg *tests.OperatorConfig
+
+		ginkgo.BeforeEach(func() {
+			ocfg = &tests.OperatorConfig{
+				Namespace:       ns,
+				ReleaseName:     "operator",
+				Image:           cfg.OperatorImage,
+				Tag:             cfg.OperatorTag,
+				ImagePullPolicy: v1.PullIfNotPresent,
+			}
+			oa = tests.NewOperatorActions(cli, c, asCli, aggrCli, apiExtCli, tests.DefaultPollInterval, ocfg, e2econfig.TestConfig, nil, fw, f)
+			ginkgo.By("Installing CRDs")
+			oa.InstallCRDOrDie(ocfg)
+			ginkgo.By("Installing tidb-operator")
+			oa.CleanOperatorOrDie(ocfg)
+			oa.DeployOperatorOrDie(ocfg)
+		})
+
+		ginkgo.AfterEach(func() {
+			ginkgo.By("Uninstall tidb-operator")
+			oa.CleanOperatorOrDie(ocfg)
+			ginkgo.By("Uninstalling CRDs")
+			oa.CleanCRDOrDie()
+		})
+
+		ginkgo.It("Deploy TidbCluster and check the result", func() {
+			ginkgo.By(fmt.Sprintf("deploy original tc %q", utilimage.TiDBV4Prev))
+			tcName := "tidbcluster1"
+			tc := fixture.GetTidbCluster(ns, tcName, utilimage.TiDBV4Prev)
+			tc.Spec.PD.Replicas = 1
+			tc.Spec.TiKV.Replicas = 1
+			tc.Spec.TiDB.Replicas = 1
+
+			err := genericCli.Create(context.TODO(), tc)
+			framework.ExpectNoError(err, "Expected TiDB cluster created")
+			err = oa.WaitForTidbClusterReady(tc, 10*time.Minute, 5*time.Second)
+			framework.ExpectNoError(err, "Expected TiDB cluster ready")
+
+			pdPods, err := getPods(labels.SelectorFromSet(label.New().Instance(tcName).PD().Labels()).String(), ns, c)
+			framework.ExpectNoError(err, "failed to get PD pods")
+
+			ginkgo.By("Set --selector=version=old for the default TiDB Operator")
+			ocfg.Selector = []string{"version=old"}
+			oa.UpgradeOperatorOrDie(ocfg)
+			log.Logf("Upgrade operator with --set-string \"selector=version=old\"")
+
+			ginkgo.By("Upgrade TidbCluster 1 version, wait for 2 minutes, check that no rolling update occurs")
+			err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+				tc.Spec.Version = utilimage.TiDBV4
+				return nil
+			})
+			framework.ExpectNoError(err, "failed to update TidbCluster 1 to upgrade PD version to %v", utilimage.TiDBV4)
+
+			err = wait.Poll(5*time.Second, 2*time.Minute, func() (done bool, err error) {
+				// confirm the TidbCluster 1 PD haven't been changed
+				changed, err := utilpod.PodsAreChanged(c, pdPods)()
+				if err != nil {
+					log.Logf("ERROR: meet error during verify TidbCluster 1 PD pods, err:%v", err)
+					return true, nil
```

Comment:
Please update the code as https://github.com/pingcap/tidb-operator/pull/3911 has merged.

Reply:
Please also check the other codes that calling the same function.

Reply:
Update